packages/firebase_auth/firebase_auth_web/lib/src/interop/auth_interop.dart

```diff
@@ -278,7 +278,7 @@ class OAuthProviderJsImpl extends AuthProviderJsImpl {
   external OAuthProviderJsImpl setCustomParameters(
     dynamic customOAuthParameters,
   );
-  external OAuthCredential credential([String? idToken, String? accessToken]);
+  external OAuthCredential credential([Object? idToken, String? accessToken]);
```

Comment:
update to match below noted API call:

```dart
return auth_interop.OAuthProvider(credential.providerId).credential(
       OAuthCredentialOptions,
      );
```

Reply:
HI @russellwheatley, thank you so much for your help! I defined an Object because I can't express an Union type String | OauthCredentialOptions in dart. The idea was to be 100% aligned with the javascript logic behind:
https://github.com/firebase/firebase-js-sdk/blob/5ad7ff2ae955c297556223e6cb3ad9d4b897f664/packages/auth/src/authcredential.js#L649

If I define an OauthCredentialOptions, then the logic branch in JS for "String" type will be disabled.
How can I define properly in the auth_interop class that the "credential" method accept both parameters types?

Reply:
@rodriguezc 

We don't need this anymore:
```dart
return OAuthProvider(oAuthCredential.providerId).credential(
    accessToken: oAuthCredential.accessToken,
    idToken: oAuthCredential.idToken as String,
  );
```

We can use this for all scenarios:
```dart
return OAuthProvider(oAuthCredential.providerId).credential(OAuthCredentialOptions);
```

This is because every property on `OAuthCredentialOptions` is optional (i.e. we can have `accessToken`, `idToken` & `rawNonce` as properties on `OAuthCredentialOptions` if we want). No need to change anything on the platform interface.

Please initialize `auth_interop.OAuthCredentialOptions` with the above three arguments that can be `nullable`. (something like: `auth_interop.OAuthCredentialOptions(accessToken, idToken,rawNonce)`)